jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -378,11 +378,13 @@ private object Watches {
     }
 
     val (trainIter1, trainIter2) = trainPoints.duplicate
-    val trainMatrix = new DMatrix(trainIter1, cacheDirName.map(_ + "/train").orNull)
+    val trainMatrix = new DMatrix(trainIter1, cacheDirName.map(_ + "/train").orNull,
+      params.get("batchSize").toString.toInt)
     val trainGroups = buildGroups(trainIter2.map(_.group).toSeq).toArray
     trainMatrix.setGroup(trainGroups)
 
-    val testMatrix = new DMatrix(testPoints.iterator, cacheDirName.map(_ + "/test").orNull)
+    val testMatrix = new DMatrix(testPoints.iterator, cacheDirName.map(_ + "/test").orNull,
```

Comment:
would like to add todo and make it backward compatible with saved model without batch_size parameter from core. TODO for next pull, make change to core and include batch_size in model serialization